pkg/cloud/api/mutable_resource.go

```diff
@@ -195,11 +195,41 @@ func (u *mutableResource[GA, Alpha, Beta]) CheckSchema() error {
 	if err != nil {
 		return err
 	}
+	ga, _ := u.ToGA()
+	alpha, _ := u.ToAlpha()
+	err = checkSubsetOf(ga, alpha)
+
+	if err != nil {
+		return fmt.Errorf("checkSubsetOf(ga, alpha) = %v, want nil", err)
+	}
+
 	err = checkSchema(reflect.TypeOf(&u.beta))
 	if err != nil {
 		return err
 	}
-	// TODOD(kl52752) Add validation that GA is a subset of Beta and Alpha.
+	beta, _ := u.ToBeta()
+	err = checkSubsetOf(ga, beta)
+
+	if err != nil {
+		return fmt.Errorf("checkSubsetOf(ga, beta) = %v, want nil", err)
+	}
+
+	return nil
+}
+
+func checkSubsetOf[T1 any, T2 any](t1 *T1, t2 *T2) error {
+	if isPlaceholderType(t2) {
+		return nil
+	}
+	diff, err := diffBetweenTypes(t1, t2)
+	if err != nil {
+		return err
+	}
+	for _, d := range diff.Items {
+		if d.State == DiffItemOnlyInA {
+			return fmt.Errorf("GA fields not present in Alpha type. Paths: %s", d.Path)
```

Comment:
error message doesn't match func 

fmt.Errorf("checkSubsetOf(): field %s is in %T but not in %T",...)

Reply:
done